cmd/kubeadm/app/phases/addons/dns/dns.go

```diff
@@ -189,9 +189,11 @@ func createCoreDNSAddon(deploymentBytes, serviceBytes, configBytes []byte, clien
 	// Assume that migration is always possible, rely on migrateCoreDNSCorefile() to fail if not.
 	canMigrateCorefile := true
 
+	configMapClient := client.CoreV1().ConfigMaps(coreDNSConfigMap.GetNamespace())
+
 	if corefile == "" || migration.Default("", corefile) {
 		// If the Corefile is empty or default, the latest default Corefile will be applied
-		if err := apiclient.CreateOrUpdateConfigMap(client, coreDNSConfigMap); err != nil {
+		if err := apiclient.CreateOrUpdate(context.TODO(), configMapClient, coreDNSConfigMap); err != nil {
```

Comment:
I used `TODO()` in places where a context might be provided externally in future, I can switch them all to `Background()` if that’s more appropriate.

Reply:
Unlike other kubernetes components which fetch logger from a context to support constructional logging, kubeadm doesn't support constructional logging, so `context.Background()` is preferred. cc @neolit123 what do you think?

Reply:
yes, plese use background